### PR TITLE
[daydayup_list_1] learing list

### PR DESCRIPTION
**Pull request by xuehuasu**
### What?

- brief: list的学习
- author: xuehuasu , ivarn-d , tigerwe

### What changes are included in this PR?

三位组员的学习笔记以及学习过程的代码

### Are these changes tested?

三位组员均尝试编写list.h的简化版，其中`ivarn-d`和`tigerwe`的代码仍有问题，还需要改进

### CODE REVIEW

File: stl/daydayup/ivarn-d/src/task_1/main.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    itrBegin++;
+  }
+
+  std::cout << std::endl;
+}
+
+int main(int argc, char* argv[]) {
+  std::cout << "Test bdgin !" << std::endl;
+  // test int
+  Yami::list<int> list_INT;
```
Comment (Light-City) [inline stl/daydayup/ivarn-d/src/task_1/main.cpp:28]: follow google var rule.

File: stl/daydayup/ivarn-d/src/task_1/main.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  list_INT.push_back(3);
+  list_INT.push_back(4);
+  list_INT.push_back(5);
+  list_INT.push_back(2);
+
+  printfList(list_INT);
+
+  std::cout << "delete nums: " << list_INT.remove(2) << std::endl;
+
+  printfList(list_INT);
```
Comment (Light-City) [inline stl/daydayup/ivarn-d/src/task_1/main.cpp:40]: ditto

File: stl/daydayup/ivarn-d/src/task_1/main.cpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+  std::cout << "size: " << list_INT1.size() << std::endl;
+
+  printfList(list_INT1);
+
+  // test class  主要看一下资源析构情况
+  Test test1;
+  Test test2;
+  Test test3;
+  Yami::list<Test> list_CLASS;
```
Comment (Light-City) [inline stl/daydayup/ivarn-d/src/task_1/main.cpp:67]: ditto

File: stl/daydayup/tigerwe/src/task_1/main.cpp [hunk 4]
```diff
@@ -0,0 +1,96 @@
+#include <iostream>
+#include "stl_list.h"
```
Comment (Light-City) [inline stl/daydayup/tigerwe/src/task_1/main.cpp:2]: Why are your tests the same as those of the students above?

File: stl/daydayup/tigerwe/src/task_1/stl_list.h [hunk 5]
```diff
@@ -0,0 +1,238 @@
+#include "stl_def.h"
+
+/** @file stl_list.h
+ *
+ *  This is an stl_list header file, implement double loop list warppes
+ *
+ *  Created by yejy on 18-8-18
+ *  copyright (c) yejy. all rights reserved
```
Comment (Light-City) [inline stl/daydayup/tigerwe/src/task_1/stl_list.h:8]: Plagiarize other people's homework without changing the header?

File: stl/daydayup/xuehuasu/src/task_1/mlist.h [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    head = new List_node_base();
+    head->next = head;
+    head->prev = head;
+  }
+  void print() {
+    iterator begin = this->begin();
+    iterator end = this->end();
+    while (begin != end) {
+      node* temp = static_cast<node*>(begin.cur);
+      printf("%d ", (int)(temp->data));
```
Comment (Light-City) [inline stl/daydayup/xuehuasu/src/task_1/mlist.h:131]: use std::cout

File: stl/daydayup/xuehuasu/src/task_1/mlist.h [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+};
+
+template <typename _Tp>
+struct List {
+  typedef List_iterator<_Tp> iterator;
+  typedef List_node<_Tp> node;
+  typedef List_node_base basenode;
+
+  List() { init(); }
+  List(const List<_Tp>& t) {
```
Comment (Light-City) [inline stl/daydayup/xuehuasu/src/task_1/mlist.h:75]: missing destructor？

**Comment by xuehuasu** [comment]
我一定督促队员们禁止抄袭

**Pull request closed without merging**